application/server/execute_command_test.go

```diff
@@ -234,3 +237,94 @@ func Test_TrustWorkspaceFolders(t *testing.T) {
 		assert.Contains(t, c.TrustedFolders(), types.FilePath("/path/to/folder2"))
 	})
 }
+
+// waitForCancelTestCommand is a test command that waits until its context is canceled.
+type waitForCancelTestCommand struct {
+	command     types.CommandData
+	t           *testing.T
+	loc         server.Local
+	cmdCanceled chan bool
+}
+
+func newWaitForCancelTestCommand(t *testing.T, loc server.Local, cmdCanceled chan bool) *waitForCancelTestCommand {
+	t.Helper()
+	return &waitForCancelTestCommand{
+		command: types.CommandData{
+			CommandId: "internal.waitForCancelTestCommand",
+			Title:     "Wait For Cancel Test Command",
+		},
+		t:           t,
+		loc:         loc,
+		cmdCanceled: cmdCanceled,
+	}
+}
+
+func (w *waitForCancelTestCommand) Command() types.CommandData {
+	return w.command
+}
+
+func (w *waitForCancelTestCommand) Execute(ctx context.Context) (any, error) {
+	w.t.Log("waitForCancelTestCommand: Execute started.")
+
+	// Command ID should always be 1 (as a number!), as it is the first command we run on the fake test server.
+	cancelParams := sglsp.CancelParams{ID: sglsp.ID{Num: 1, IsString: false}}
+	err := w.loc.Client.Notify(context.Background(), "$/cancelRequest", cancelParams)
+	assert.NoError(w.t, err, "Failed to send $/cancelRequest notification")
+
+	w.t.Log("waitForCancelTestCommand: Entering wait on ctx.Done().")
+	<-ctx.Done()
+	w.t.Logf("waitForCancelTestCommand: ctx.Done() returned. ctx.Err() is: %v\n", ctx.Err())
+	w.cmdCanceled <- errors.Is(ctx.Err(), context.Canceled)
```

Comment:
this part should be part of the test, not the command, I think. the command should just wait, until the context is cancelled.